extensions/amp-form/0.1/amp-form.js

```diff
@@ -402,17 +410,15 @@ export class AmpForm {
    * Handles submissions through action service invocations.
    *   e.g. <img on=tap:form.submit>
    * @param {!../../../src/service/action-impl.ActionInvocation} invocation
+   * @return {?Promise}
    * @private
    */
   handleSubmitAction_(invocation) {
     if (this.state_ == FormState.SUBMITTING || !this.checkValidity_()) {
-      return;
+      return null;
```

Comment:
It's an antipattern for a function that returns a Promise to also return a synchronous value. It prevents chaining from working correctly, and mixes synchronous and asynchronous behavior ([which releases Zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony)).

Bad:
```js
/**
 * The major drawbacks here are:
 * 1. You can forget to check if the result is null and have a null reference error
 * 2. The function sometimes returns synchronously and sometimes asynchronously, 
 *      so timing expectations are weird
 * @param {boolean} flag
 * @return {?Promise}
 */
function doStuff(flag) { 
  if (flag) {
    return Promise.resolve(successValue);
  } else {
    return null;
  }
}

const result = doStuff(false);
if (result) {
  doStuff().then(() => doMoreStuff());
}
```

Good:
```js
function doStuff(flag) {
  if (flag) {
    Promise.resolve(successValue);
  } else {
    Promise.resolve(null);
  }
}

const result = doStuff(false).then(result => {
  if (result) { 
    return doMoreStuff();
  }
});
```

Reply:
I've learned more from this PR than my 5 years in college during my CS degree 😂 

Thank you! Will make the changes 😄